src/main/java/Parser.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Class to handle parsing of user input.
+ */
+public class Parser {
+
+    /**
+     * Given a line, obtains the command that was present in that line.
+     * 
+     * @param line The line input by the user
+     * @return Command object representing the command that was present in the line
+     * @throws InvalidCommandException If there is no valid Command in the line
+     */
+    public Command getCommandFromLine(String line) throws InvalidCommandException {
+        return Command.getFromString(line.split(" ")[0].trim());
+    }
+
+    /**
+     * Given a line, obtains the (numerical) index present in the line.
+     * 
+     * @param line The line input by the user
+     * @return The int index present in the line
+     */
+    public int getIndexFromLine(String line) {
+        try {
+            return Integer.parseInt(line.split(" ")[1]) - 1;
+        } catch (Exception e) {
```

Comment:
I think it would be much better to catch specific exceptions instead of catching all of them!